kernel/posix/pthread.c

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <atomic.h>
+#include "ksched.h"
+#include "wait_q.h"
+
+/* Kernel Structure for Number of threads */
+struct k_thread threads[CONFIG_MAX_PTHREAD_COUNT];
+
+static int is_prio_valid(int priority)
+{
+	if ((priority < K_HIGHEST_APPLICATION_THREAD_PRIO)
+	    || (priority > K_HIGHEST_APPLICATION_THREAD_PRIO)) {
+		return -1;
+	} else       {
+		return 0;
+	}
+
+}
+
+int pthread_attr_setschedparam(pthread_attr_t *attr,
```

Comment:
Please add all these functions also to posix_cheats.h
Otherwise you prevent the POSIX arch port from using the host ones.